src/response_template.rs

```diff
@@ -107,6 +107,51 @@ impl ResponseTemplate {
     /// Set the response body with bytes.
     ///
     /// It sets "Content-Type" to "application/octet-stream".
+    ///
+    /// To set a body with bytes but a different "Content-Type"
+    /// [`set_body_raw`](#method.set_body_raw) can be used.
+    ///
+    /// ### Example:
+    /// ```rust
+    /// use surf::mime;
+    /// use wiremock::{MockServer, Mock, ResponseTemplate};
+    /// use wiremock::matchers::method;
+    ///
+    /// mod external {
+    ///     // This could be a method of a struct that is
+    ///     // implemented in another crate and the struct
+    ///     // does not implement Serialize.
+    ///     pub fn body() -> Vec<u8>{
+    ///         r#"{"hello": "world"}"#.as_bytes().to_owned()
+    ///     }
+    /// }
+    ///
+    /// #[async_std::main]
+    /// async fn main() {
+    ///     // Arrange
+    ///     let mock_server = MockServer::start().await;
+    ///     let template = ResponseTemplate::new(200).set_body_raw(
+    ///         external::body(),
+    ///         "application/json"
+    ///     );
+    ///     Mock::given(method("GET"))
+    ///         .respond_with(template)
+    ///         .mount(&mock_server)
+    ///         .await;
+    ///
+    ///     // Act
+    ///     let mut res = surf::get(&mock_server.uri())
+    ///         .await
+    ///         .unwrap();
+    ///     let body = res.body_string()
+    ///         .await
+    ///         .unwrap();
+    ///
+    ///     // Assert
+    ///     assert_eq!(body, r#"{"hello": "world"}"#);
+    ///     assert_eq!(res.mime(), Some(mime::APPLICATION_JSON));
+    /// }
+    /// ```
```

Comment:
The example is perfect, but it should be on `set_body_raw` I'd say

Reply:
:scream: fixed.

Think for all the useful feedback :+1: .